### PR TITLE
Uri converter

### PR DESCRIPTION
**Pull request by johnament**
This introduces a new built in required URI converter.  This is needed by rest client.

I did a separate commit to clean up the formatting on `ArrayConverterBean`, it was a bit hard to follow.

### CODE REVIEW

**Comment by Emily-Jiang** [comment]
The URI converter should be already covered by the default converter as it has a constructor taking a single string. The next step I am going to do is to remove all built in converters instead of adding any more.

**Comment by johnament** [comment]
Looking at github, I don't see your requested changes.  But you want to revert the javadoc and spec changes in favor of claiming that they just work because of implicit converters?

The difference I think is that `URI.create`throws an `IllegalArgumentException` whereas `new URI()` throws `URISyntaxException`.

**Comment by Emily-Jiang** [comment]
@johnament for implementation, the call new URI(string) etc should catch the exception and wrap up with IllegalArgumentException, which should be already there in the part of implicit converter support.

**Comment by struberg** [comment]
+1 for relying on implicit converter in this case.
